dot/sync/service.go

```diff
@@ -119,6 +135,11 @@ func NewSyncService(cfgs ...ServiceConfig) *SyncService {
 		waitPeersDuration:     waitPeersDefaultTimeout,
 		stopCh:                make(chan struct{}),
 		seenBlockSyncRequests: lrucache.NewLRUCache[common.Hash, uint](100),
+		workerPool: NewWorkerPool(WorkerPoolConfig{
+			MaxRetries: 5,
+			// TODO: This should depend on the actual configuration of the currently used sync strategy.
+			Capacity: defaultNumOfTasks * 10,
```

Comment:
Why times 10?

Reply:
Just an arbitrary value that "should be enough" while I was testing the branch on the Westend genesis instance. Apart from wanting to do this testing, this TODO and the other one about back pressure are the reasons why this is still a draft.
Do you think it makes sense to use the actual value for number of tasks configured in the sync strategy? And should we add some extra space here or not?

Reply:
Ok got it, we can move it to a constant to self explain the usage.

Reply:
I'd like to try and make this depend on the configuration of the strategy. But if that's too invasive, I'll move it to a constant.

Reply:
I've done this in [07358f2](https://github.com/ChainSafe/gossamer/pull/4258/commits/07358f22b7c0e923b10a695c0d41c43a31bf0f6b). It required extending the `Strategy` interface and delaying initialization of the worker pool.